Update README with more details (#188)
sigstore/timestamp-authority: README.md

Timestamps conform to the [RFC 3628 policy](https://datatracker.ietf.org/doc/html/rfc3628).

## Security model

[Trusted timestamping](https://en.wikipedia.org/wiki/Trusted_timestamping) is a process that has been around for some time.
It provides a timestamp record of when a documented was created or modified.

A timestamp authority creates signed timestamps using public key infrastructure. The operator of the timestamp
authority must secure the signing key material to prevent unauthorized timestamp signing.

A timestamp authority should also verify its own clock. We provide a configuration to periodically check the current
time against well-known NTP sources.

## Timestamping within Sigstore

Timestamps are a critical component of [Rekor](https://github.com/sigstore/rekor), Sigstore's signature transparency log.
Timestamps are used to verify short-lived certificates. Currently, the timestamp comes from Rekor's own internal clock,

> which is not externally verifiable. Using signed timestamps issued from timestamp authorities mitigates the risk of

Reviewer: ```suggestion
which is not externally verifiable or immutable. Using signed timestamps issued from timestamp authorities mitigates the risk of
```

Rekor's clock being manipulated.

As a artifact signer, you can:

* Generate a signature over an artifact

> * Fetch a timestamp for that signature (more below in "What to sign")

Reviewer: ```suggestion
* Fetch a timestamp for that signature (more below in [What to sign](##what-to-sign))
```

I think, vscode will have autocomplete for it

* Upload the signature, artifact hash, and certificate to Rekor (hashedrekord record type)
* Upload the timestamp to Rekor (rfc3161 record type)

>    * This step is important because it makes the timestamps publicly auditable

Reviewer: In future work, will timestamp-authority handle this itself?

Author: Someone asked about this awhile ago. My thought was to make this always client-driven, but maybe we make it configurable? What do you think?

As an artifact verifier:

* Fetch the artifact entry from Rekor
* If the artifact was signed with a certificate, verify its expiration
   * If you trust Rekor's clock, verify the certificate with the timestamp in the Rekor response
   * If you trust an external timestamp authority, fetch the timestamp from Rekor, verify the
     signed timestamp, and verify the certificate using the signed timestamp

### What to sign

> For usage within Sigstore, we recommend signing over a value that is associated with an artifact.

Reviewer: ```suggestion
For usage within Sigstore, we recommend signing over a value that is associated with a signature.
```

For [Cosign](https://github.com/sigstore/cosign/), we have chosen to sign the artifact signature,

> a process called "countersigning". We sign over the raw signature, not a base64-encoded value. Signing

Reviewer: ```suggestion
a process called "countersigning". We sign over the raw signature bytes, not a base64-encoded value. Signing
```

> over the signature ensures that the signature, not the artifact, was created a certain time.

Reviewer: ```suggestion
over the signature ensures that the signature, not the artifact, was created at a certain time.
```
